tests/foreman/virtwho/test_ui_virtwho.py

```diff
@@ -420,3 +428,201 @@ def test_positive_virtwho_configs_widget(session, form_data):
+        user = session.user.read(username)
+        assert user['roles']['resources']['assigned'] == ['Virt-who Reporter']
+        restart_virt_who_service()
+        assert get_virtwho_status() == 'running'
+        with Session(test_name, username, password) as newsession:
+            try:
+                newsession.virtwho_configure.read(virt_who_name)
+            except Exception:
+                pass
+            else:
+                raise VirtWhoError(
+                    'Virt-who Reporter should not have the permission to read config')
+        session.user.delete(username)
+        assert not session.user.search(username)
+
+
+def test_positive_virt_who_viewer_permission(session, test_name, form_data):
+    """Verify the virt-who viewer role can TRULY work.
+
+    :id: 9cdd3bbe-cfb2-4297-92fb-110515f835b3
+
+    :expectedresults:
+        Virt-who Viewer Role granting permissions to see all configurations
+        including their configuration scripts, which means viewers could still
+        deploy the virt-who instances for existing virt-who configurations.
+    """
+    username = gen_string('alpha')
+    password = gen_string('alpha')
+    with session:
+        # Create an user
+        session.user.create({
+            'user.login': username,
+            'user.mail': 'test@test.com',
+            'user.auth': 'INTERNAL',
+            'user.password': password,
+            'user.confirm': password,
+        })
+        # Create a virt-who config plugin
+        virt_who_name = gen_string('alpha')
+        form_data['name'] = virt_who_name
+        session.virtwho_configure.create(form_data)
+        values = session.virtwho_configure.read(virt_who_name)
+        command = values['deploy']['command']
+        deploy_configure_by_command(command, debug=True)
+        assert session.virtwho_configure.search(virt_who_name)[0]['Status'] == 'ok'
+        # Check the permissioin of Virt-who Viewer
+        session.user.update(
+            username,
+            {'roles.resources.assigned': ['Virt-who Viewer']}
+        )
+        user = session.user.read(username)
+        assert user['roles']['resources']['assigned'] == ['Virt-who Viewer']
+        with Session(test_name, username, password) as newsession:
+            newsession.virtwho_configure.read(virt_who_name)
+            try:
+                viewer_virt_who_name = gen_string('alpha')
+                form_data['name'] = viewer_virt_who_name
+                newsession.virtwho_configure.create(form_data)
+            except Exception:
+                pass
+            else:
+                raise VirtWhoError(
+                    'Virt-who Viewer should not have the permission to creat config')
+        # Delete the created user
+        session.user.delete(username)
+        assert not session.user.search(username)
+
+
+def test_positive_virt_who_manager_permission(session, test_name, form_data):
```

Comment:
rename the case to test_positive_virtwho_manager_role

Reply:
ok